partiql-spi/src/main/java/org/partiql/eval/value/Datum.java

```diff
@@ -576,69 +578,77 @@ static Datum doublePrecision(double value) {
 
     @Deprecated
     @NotNull
-    static Datum decimalArbitrary(@NotNull BigDecimal value) {
-        return new DatumDecimal(value, PType.typeDecimalArbitrary());
+    static Datum decimal(@NotNull BigDecimal value) {
+        return new DatumDecimal(value, PType.decimal());
     }
 
     @NotNull
     static Datum decimal(@NotNull BigDecimal value, int precision, int scale) {
-        return new DatumDecimal(value, PType.typeDecimal(precision, scale));
+        return new DatumDecimal(value, PType.decimal(precision, scale));
     }
 
+    // CHARACTER STRINGS
+
     @NotNull
-    static Datum bool(boolean value) {
-        return new DatumBoolean(value);
+    static Datum string(@NotNull String value) {
+        return new DatumString(value, PType.string());
     }
 
     @NotNull
-    static Datum sexp(@NotNull Iterable<Datum> values) {
-        return new DatumCollection(values, PType.typeSexp());
+    static Datum symbol(@NotNull String value) {
+        return new DatumString(value, PType.symbol());
     }
 
     @NotNull
-    static Datum list(@NotNull Iterable<Datum> values) {
-        return new DatumCollection(values, PType.typeList());
+    static Datum clob(@NotNull byte[] value) {
+        return new DatumBytes(value, PType.clob(Integer.MAX_VALUE));
     }
 
+    // BYTE STRINGS
+
     @NotNull
-    static Datum struct(@NotNull Iterable<Field> values) {
-        return new DatumStruct(values);
+    static Datum blob(@NotNull byte[] value) {
+        return new DatumBytes(value, PType.blob(Integer.MAX_VALUE));
     }
 
+    // DATE/TIME
+
     @NotNull
-    static Datum string(@NotNull String value) {
-        return new DatumString(value, PType.typeString());
+    static Datum date(@NotNull Date value) {
+        return new DatumDate(value);
     }
 
     @NotNull
-    static Datum symbol(@NotNull String value) {
-        return new DatumString(value, PType.typeSymbol());
+    static Datum timeWithoutTZ(@NotNull Time value) {
```

Comment:
From your naming in `PType`, I would've thought that the static method would be called `time` and `timez`.

Reply:
It should've been, I need to update.